### DISCUSSION TITLE
Difficulty configuring MSE

### DISCUSSION THREAD

**Discussion by seIIers** [Megghy/MultiSEngine#9] [Q&A] [answered]
I'm having trouble setting up MSE for two TShock version 4.5.17 servers on the same VPS running Ubuntu 20.04. My config files can be found [here](https://gist.github.com/sellerscrisp/7db484c97eb6bf2b8eb920265d8e45fa). Initially, I was using supervisord to manage each server instance, but I figured there may be some complications. Both servers run perfectly when started initially, and a reload in the console prints `<MultiSCore> Read config success.` but I cannot run any `/mse` related commands in the server or the console. 

I'm happy to provide other information upon request--I'm quite new to TShock and was having a bit of trouble both understanding and eventually accessing the MSE documentation. Any help would be greatly appreciated!

**Comment by Megghy** [comment, marked as answer]
mse is an application independent of tshock, while msc is a plugin based on tshock, these two are different things, if you use mse then you do not need to install msc, and vice versa

**Reply by seIIers** [reply]
My apologies, I didn't realize how amazing this application is (I thought it was just plugin specific). I ended up switching to MSE and it worked perfectly. 

However, the server with a premade world had some issues around the spawn point--it looks like there has been a fairly large chunk taken out near the spawning area. See [server world](https://dl.dropboxusercontent.com/s/fj5eeh56k605ral/3prIidKrdW_Terraria_A_good_day_to_dig_hard.png) vs. [singleplayer world](https://dl.dropboxusercontent.com/s/xo5x8pb83ia98ls/hUQ8xUAOYz_Terraria_A_good_day_to_dig_hard.png)

I should likely create an issue but am hoping you might be aware of the issue already. Is there a specific way to clear the MSE "cache" or store for the server's world? I tried going through the source code but am completely lost when it comes to this complex programming. I've tried changing the world file in the TShock instance but nothing changes. I have also attempted to change the world spawn before replacing the file but that does not seem to work either.

**Reply by Megghy** [reply]
Can you attach the runtime environment or any information about the error reported? I tested it in 1.4.3.6 and found no problems

**Reply by seIIers** [reply]
``` 
[Warn] Server recieve loop abnormally terminated.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at TrProtocol.Models.SectionData.SectionDataSerializer.<_Read>g__deserializeTile|0_1(BinaryReader br)
   at TrProtocol.Models.SectionData.SectionDataSerializer.<_Read>g__deserialize|0_0(BinaryReader br, <>c__DisplayClass0_0& )
   at TrProtocol.Models.SectionData.SectionDataSerializer._Read(BinaryReader reader)
   at TrProtocol.FieldSerializer`1.Read(BinaryReader br)
   at TrProtocol.PacketSerializer.<>c__DisplayClass10_1.<GenerateSerializers>b__5(Object o, BinaryReader br)
   at TrProtocol.PacketSerializer.<>c__DisplayClass11_0.<RegisterPacket>b__2(BinaryReader br)
   at TrProtocol.PacketSerializer.Deserialize(BinaryReader br0)
   at MultiSEngine.Core.Adapter.BaseAdapter.RecieveLoop()
```

**Reply by seIIers** [reply]
I should also mention the world is slowly being put back together. Background tiles first, then block tiles.

**Reply by Megghy** [reply]
Because mse is actually no longer processing packets from the specified server when this error message is displayed, so packets from the old server will not be intercepted, although normally the player should be sent back to the original server
It looks like there is some problem with the protocol library, I will look at it over the weekend

**Reply by seIIers** [reply]
Thank you for your help. I wish I could offer mine!

**Reply by Megghy** [reply]
Are you using the version in the release or the single file version generated by ci? If it's release, you can try replacing the file with the same name in the root directory with this file
[TrProtocol.zip](https://github.com/Megghy/MultiSEngine/files/8644467/TrProtocol.zip)

**Reply by seIIers** [reply]
That worked, thank you! Guess I need to use the CI generated version.

**Reply by Megghy** [reply]
The ci generation is complete
